### PR TITLE
chore(explorer): Refactor it to make it reusable for `exp/query`.

### PR DESCRIPTION
**Pull request by kavirajk**
### What's being changed:

Changes:
1. Inject all it's dependencies explicitly.
1. Remove unused `hybridSearcher` interface in explorer
1. Split `metrics` and `config` to accept only what it uses
1. Decouple it from huge `uc.SchemaGettter`. As explorer only uses two methods from it.
2. Set the `schemaGetter` inside the constructor itself and avoid needing to set explicitly via `SetSchemaGetter`. This makes the returned `Explorer` after `NewExplorer` consistent and read-to-use without needing any additional steps.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: usecases/traverser/metrics.go [hunk 1]
```diff
@@ -116,20 +112,3 @@ func (m *Metrics) QueriesGetDec(className string) {
 		"query_type": "get_graphql",
 	}).Dec()
 }
-
-func (m *Metrics) AddUsageDimensions(className, queryType, operation string, dims int) {
```
Comment (kavirajk) [inline usecases/traverser/metrics.go:120]: this is only used in `explorer` so moved it from `traverser` -> `explorer` method.

File: usecases/monitoring/prometheus.go [hunk 2]
```diff
@@ -59,17 +59,22 @@ type PrometheusMetrics struct {
...
-	BackupRestoreClassDurations         *prometheus.SummaryVec
-	BackupRestoreBackupInitDurations    *prometheus.SummaryVec
-	BackupRestoreFromStorageDurations   *prometheus.SummaryVec
-	BackupRestoreDataTransferred        *prometheus.CounterVec
-	BackupStoreDataTransferred          *prometheus.CounterVec
+
+	// Q(Kavi): Why need two metrics `QueryDimensions` and `QueryDimensionsCombined`.
+	// seeing their usage, suming `QueryDimensions` without any labels should be
+	// same as `QueryDimensionsCombined`.
+	QueryDimensions         *prometheus.CounterVec
```
Comment (reyreaud-l) [inline usecases/monitoring/prometheus.go:66]: @antas-marcin or @etiennedi might have more input here, i remember seeing discussion about this recently

File: usecases/traverser/explorer.go [hunk 3]
```diff
@@ -47,18 +46,19 @@ var _NUMCPU = runtime.GOMAXPROCS(0)
...
+	searcher          ObjectsSearcher
 	logger            logrus.FieldLogger
 	modulesProvider   ModulesProvider
-	schemaGetter      uc.SchemaGetter
+	schemaGetter      SchemaGetter
 	nearParamsVector  *nearParamsVector
 	targetParamHelper *TargetVectorParamHelper
-	metrics           explorerMetrics
-	config            config.Config
+	config            *ExplorerConfig
```
Comment (reyreaud-l) [inline usecases/traverser/explorer.go:55]: Any reason in particular making them pointers ?

File: usecases/traverser/metrics.go [hunk 4]
```diff
@@ -32,11 +30,9 @@ func NewMetrics(prom *monitoring.PrometheusMetrics) *Metrics {
 	}
 
 	return &Metrics{
-		queriesCount:       prom.QueriesCount,
-		queriesDurations:   prom.QueriesDurations,
-		dimensions:         prom.QueryDimensions,
```
Comment (reyreaud-l) [inline usecases/traverser/metrics.go:37]: Were the dimensions unused ?

**Review by reyreaud-l — DISMISSED** [review]

**Pull request closed without merging**